pkgs/applications/networking/mailreaders/claws-mail/default.nix

```diff
@@ -1,97 +1,171 @@
+, enableEnchant ? enableSpellcheck, enchant
+, enableDbus ? true, dbus, dbus-glib
+, enableLdap ? true, openldap
+, enableNetworkManager ? true, networkmanager
+, enableLibetpan ? true, libetpan
+, enableValgrind ? true, valgrind
+, enableSvg ? true, librsvg
+
+# Configure claws-mail's plugins
+, enablePluginAcpiNotifier ? true
+, enablePluginAddressKeeper ? true
+, enablePluginArchive ? true, libarchive
+, enablePluginAttRemover ? true
+, enablePluginAttachWarner ? true
+, enablePluginBogofilter ? true
+, enablePluginBsfilter ? true
+, enablePluginClamd ? true
+, enablePluginDillo ? true
+, enablePluginFetchInfo ? true
+, enablePluginGdata ? true, libgdata
+, enablePluginLibravatar ? enablePluginRavatar
+, enablePluginLitehtmlViewer ? true, gumbo
+, enablePluginMailmbox ? true
+, enablePluginManageSieve ? true
+, enablePluginNewMail ? true
+, enablePluginNotification ? (enablePluginNotificationDialogs || enablePluginNotificationSounds), libcanberra-gtk2, libcanberra-gtk3, libnotify
+, enablePluginPdfViewer ? enablePluginPdf, poppler
+, enablePluginPerl ? true, perl
+, enablePluginPython ? true
+, enablePluginPgp ? enablePgp, gnupg, gpgme
+, enablePluginRssyl ? true, libxml2
+, enablePluginSmime ? true
+, enablePluginSpamassassin ? true
+, enablePluginSpamReport ? true
+, enablePluginTnefParse ? true, libytnef
+, enablePluginVcalendar ? true, libical
 }:
 
+
 with stdenv.lib;
 
-stdenv.mkDerivation rec {
-  pname = "claws-mail";
+let
   version = "3.17.7";
 
-  src = fetchurl {
-    url = "https://www.claws-mail.org/download.php?file=releases/claws-mail-${version}.tar.xz";
-    sha256 = "1j6x09621wng0lavh53nwzh9vqjzpspl8kh5azh7kbihpi4ldfb0";
+  # The official release uses gtk2 and contains the version tag.
+  gtk2src = {
+    rev = version;
+    sha256 = "0rqyr7f23sjpq8sy1jlvmq1ag4pzvxpqpa3pqss4gazmlf73gs3d";
   };
 
+  # The corresponding commit in the gtk3 branch. There is no git tag.
+  gtk3src = {
+    rev = "a5741530b9ab16224c64c2a5a6af79b0bf671454";
+    sha256 = "13705rw40hl3j3lkfrlzjvj00yj7lng6ljqd4vprwzkq3qm5gnbs";
+  };
+in stdenv.mkDerivation rec {
+  pname = "claws-mail";
+  inherit version;
+
+  src = fetchgit ({
+    url = "git://git.claws-mail.org/claws.git";
+  } // (if useGtk3 then gtk3src else gtk2src));
+
   outputs = [ "out" "dev" ];
 
-  patches = [
-    ./mime.patch
-  ];
+  patches = [ ./mime.patch ];
 
   preConfigure = ''
     # autotools check tries to dlopen libpython as a requirement for the python plugin
     export LD_LIBRARY_PATH=$LD_LIBRARY_PATH''${LD_LIBRARY_PATH:+:}${python}/lib
+    # generate version without .git
+    [ -e version ] || echo "echo ${version}" > version
   '';
 
   postPatch = ''
     substituteInPlace src/procmime.c \
         --subst-var-by MIMEROOTDIR ${shared-mime-info}/share
   '';
 
-  nativeBuildInputs = [ autoreconfHook pkgconfig wrapGAppsHook python.pkgs.wrapPython ];
+  nativeBuildInputs = [ autoreconfHook pkgconfig bison flex wrapGAppsHook python.pkgs.wrapPython ];
```

Comment:
How come all of a sudden?

Reply:
The tarballs contain some generated content, like precompiled translations. Those needed to be generated when compiled from `git`. The old `-gtk3` package had those inputs included and they did the job. Thus, I added them to the _merged_ package description.

Reply:
You can add it only if gtk3 is used.

Reply:
Now we are also sourcing GTK2 from git because reasons.